plugins/out_stackdriver/stackdriver.c

```diff
@@ -2834,57 +2841,61 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
         if (c->resp.status == 200) {
             ret_code = FLB_OK;
         }
-        else if (c->resp.status >= 400 && c->resp.status < 500) {
-            ret_code = FLB_ERROR;
-            /* check partial success */
-            ret_partial_success =
+        else {
+#ifdef FLB_HAVE_METRICS
+          /* check partial success */
+          ret_partial_success =
```

Comment:
[Optional] Technically, we could also guard this by response code, because you should never get a partial success response for 500 codes… Unless I'm missing something…